deepchecks/checks/performance/model_inference_time.py

```diff
@@ -0,0 +1,84 @@
+"""The model inference time check module."""
+import typing as t
+import timeit
+
+from deepchecks import SingleDatasetBaseCheck, CheckResult, Dataset, ConditionResult
+from deepchecks.utils import model_type_validation, DeepchecksValueError
+from deepchecks.string_utils import format_number
+
+
+__all__ = ['ModelInferenceTimeCheck']
+
+
+MI = t.TypeVar('MI', bound='ModelInferenceTimeCheck')
+
+
+class ModelInferenceTimeCheck(SingleDatasetBaseCheck):
+    """Measure model average inference time (in seconds) per sample."""
+
+    NUMBER_OF_SAMPLES: t.ClassVar[int] = 1000
+
+    def run(self, dataset: Dataset, model: object) -> CheckResult:
+        """Run check.
+
+        Args:
+            dataset (Dataset): samples that will be used to measure inference time
+            model (BaseEstimator): a model to measure inference time
+
+        Returns:
+            CheckResult:
+                value is of the type 'float'
+
+        Raises:
+            DeepchecksValueError: If the 'test_dataset' is not a 'Dataset' instance with a label or
+                if 'model' is not a scikit-learn-compatible fitted estimator instance
+        """
+        return self._model_inference_time_check(dataset, model)
+
+    def _model_inference_time_check(
+        self,
+        dataset: Dataset,
+        model: object # TODO: find more precise type for model
+    ) -> CheckResult:
+        check_name = type(self).__name__
+        Dataset.validate_dataset(dataset, check_name)
+        model_type_validation(model)
+
+        if dataset.features_columns() is None:
```

Comment:
We're missing dataset.validate_model(model), and it will also take care of checking that there is inference to be done

Reply:
updated